bioptim/optimization/parameters.py

```diff
@@ -273,7 +273,7 @@ def add_or_replace(ocp, _, parameter: Parameter):
             val = func(ocp, parameter.cx, **penalty.params)
             penalty.sliced_target = penalty.target
             ObjectiveFunction.ParameterFunction.clear_penalty(ocp, None, penalty)
-            ObjectiveFunction.ParameterFunction.add_to_penalty(ocp, None, val, penalty)
+            ObjectiveFunction.ParameterFunction.add_to_penalty(ocp, None, None, val, penalty)
```

Comment:
As mentioned previously, to create a new pn, we need to access to the current nlp. In some cases, I think nlp has no sense, so that it seems difficult to create a new PenaltyNodes. That is why I guess it is usefull to keep pn as None in some cases.

Reply:
Done